src/test/java/net/starschema/clouddb/jdbc/CommonTestsForResultSets.java

```diff
@@ -0,0 +1,274 @@
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class CommonTestsForResultSets {
+
+  private Connection connection;
+
+  @Before
+  public void connectForCommonTests() throws SQLException, IOException {
+    connection =
+        ConnectionFromResources.connect("installedaccount1.properties", "&useLegacySql=false");
+  }
+
+  @After
+  public void closeConnectionForCommonTests() throws SQLException {
+    connection.close();
+  }
+
+  /**
+   * A {@link java.util.function.Consumer} that can throw a {@link SQLException}
+   *
+   * @param <T> The return value's type
+   */
+  private interface ConsumerWithSQLException<T> {
+
+    void accept(T arg) throws SQLException;
+  }
+
+  /**
+   * Create a {@link Statement} that returns the appropriate result set.
+   *
+   * @param connection the connection from which to create the statement
+   * @return a statement used by common tests
+   * @throws SQLException if the {@link Connection} raises one
+   */
+  protected abstract Statement createStatementForCommonTests(final Connection connection)
+      throws SQLException;
+
+  /**
+   * Assert something about the single row and column returned by a query.
+   *
+   * @param query must result in a single row and column
+   * @param check a Consumer that can throw a {@link SQLException} - put your assertions about the
+   *     result set in this
+   * @throws SQLException whenever check throws it
+   */
+  private void assertSingleRowAndColumn(
+      final String query, final ConsumerWithSQLException<ResultSet> check) throws SQLException {
+    try (Statement stmt = createStatementForCommonTests(connection)) {
+      final ResultSet result = stmt.executeQuery(query);
+      Assertions.assertThat(result.next()).isTrue();
+      check.accept(result);
+      Assertions.assertThat(result.next()).isFalse();
+    }
+  }
+
+  @Test
+  public void testFindColumn() throws SQLException {
+    assertSingleRowAndColumn(
+        "SELECT 2 AS two", rs -> Assertions.assertThat(rs.findColumn("two")).isEqualTo(1));
+  }
+
+  /**
+   * A {@link java.util.function.Function} that can throw a {@link SQLException}
+   *
+   * @param <T> the argument's type
+   * @param <R> the returned value's type
+   */
+  private interface FunctionWithSQLException<T, R> {
+    R apply(T arg) throws SQLException;
+  }
+
+  private void assertReaderAsString(
+      final String query,
+      final FunctionWithSQLException<ResultSet, Reader> getter,
+      final String value)
+      throws SQLException {
+    assertSingleRowAndColumn(
+        query,
+        rs -> {
+          final Reader reader = getter.apply(rs);
+          final String contents =
+              new BufferedReader(reader).lines().collect(Collectors.joining(""));
+          Assertions.assertThat(contents).isEqualTo(value);
+        });
+  }
+
+  @Test
+  public void testGetAsciiStreamByLabel() throws SQLException {
```

Comment:
I tried to make sure these tests are aligned with how other tests in this repo are written but it's riddled with inconsistency 😵

Reply:
The other tests in the repo, not yours

Reply:
We can clean this thing up one PR at a time 🫡